src/copas.lua

```diff
@@ -363,11 +363,18 @@ end
 --       skt:send(body)      --> continues unencrypted
 -- @param skt Regular LuaSocket CLIENT socket object
 -- @param sslt Table with ssl parameters
+-- @param hostname SNI hostname/map
+-- @param strict Fails the handshake if the SNI host name does not exist
 -- @return wrapped ssl socket, or throws an error
-function copas.dohandshake(skt, sslt)
+function copas.dohandshake(skt, sslt, hostname, strict)
   ssl = ssl or require("ssl")
   local nskt, err = ssl.wrap(skt, sslt)
   if not nskt then return error(err) end
+  if type(hostname) == 'table' then
+    nskt:sni(hostname, strict)
+  else if hostname then
```

Comment:
the `else if` should be `elseif` probably, see the CI failure;
```
lua: error loading module 'copas' from file 'src/copas.lua':
	src/copas.lua:836: 'end' expected (to close 'function' at line 371) near '<eof>'
stack traceback:
	[C]: ?
	[C]: in function 'require'
	tests/largetransfer.lua:9: in main chunk
	[C]: ?
Makefile:40: recipe for target 'test' failed
```

Is this `if` statement needed at all? can't it just be;
```lua
  if hostname then
    nskt:sni(hostname, strict)
  end
```
or does LuaSec bork on the extra `nil` passed in, in case there is no `strict` provided?

Reply:
pushed a fix for that one